src/tool/subcommands/api_cmd/report.rs

```diff
@@ -0,0 +1,437 @@
+// Copyright 2019-2025 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+use super::ReportMode;
+use crate::rpc;
+use crate::rpc::FilterList;
+use ahash::{HashMap, HashMapExt};
+use serde::{Deserialize, Serialize};
+use similar::{ChangeTag, TextDiff};
+use std::path::Path;
+use std::time::Instant;
+use tabled::{builder::Builder, settings::Style};
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+struct PerformanceMetrics {
+    total_duration_ms: u128,
+    average_duration_ms: u128,
+    min_duration_ms: u128,
+    max_duration_ms: u128,
```

Comment:
Can we use strongly typed `Duration`? I'd avoid plain integers if there's a more expressive and feature-rich type.